### ISSUE TITLE
allow multiple device selection in reports/replay module

### ISSUE THREAD

**Issue by interduo** [traccar/traccar-web#1054]
**Is your feature request related to a problem? Please describe.**
Yes. Now we are not allow to view enought data on screen / few devices in some reports/replay module.

**Describe the solution you'd like**
Allow to choose few devices in (not always only one) in:
http://traccar:8082/reports/event
http://traccar:8082/reports/trip
http://traccar:8082/reports/stop
http://traccar:8082/reports/chart
http://traccar:8082/replay
in device selection.

So in shortcut: please change select to multiselect in all elements with labels: "Device".

**Comment by tananaev** [comment]
You have to provide some justification why we should add it. There was a reason for not implementing it - performance. We need to know why it is so important, what alternatives you considered etc.

By the way, I'm sure this has been discussed before. Have you checked existing tickets to make sure it's not a duplicate?

**Comment by dough29** [comment]
I also have that need.

On legacy UI I used to make a report for part - not all - of my devices to know what time they entered a perimeter and export the report to Excel file.

Thanks.

**Comment by tananaev** [comment]
Which report?

**Comment by dough29** [comment]
Like this :

![image](https://user-images.githubusercontent.com/2735305/205329054-334c0111-ce4c-4fb2-9d04-04701c1a8328.png)

![image](https://user-images.githubusercontent.com/2735305/205328942-e5cfc27b-a1b9-45c0-ba3a-d09edb6d955b.png)

**Comment by tananaev** [comment]
OK, so the event report?

**Comment by dough29** [comment]
In my case yes for event report, seems like OP looks for more places for this feature.

**Comment by interduo** [comment]
> You have to provide some justification why we should add it. There was a reason for not implementing it - performance. We need to know why it is so important, what alternatives you considered etc.

You say performance - it's better to send one request and get all needed data then sending few and few more requests for comparing between tracks/cars.

If You got many cars with many tracks per day in one region it's very helpful to see all tracks/cars in one screen to see all the shared parts.

> By the way, I'm sure this has been discussed before. Have you checked existing tickets to make sure it's not a duplicate?

I didn't find duplicated issues searching by title. Maybe some message about this is inside issue that the subject is not connected with the subject?

**Comment by tananaev** [comment]
> You say performance - it's better to send one request and get all needed data then sending few and few more requests for comparing between tracks/cars.

Technically there's no difference. We actually request devices one by one on the backend anyway.

But I'm not talking about theoretical performance. I'm talking about practice. For example, many people just select a group of devices when they actually don't need all devices from that group. Just because convenience.

Another very common situation is when people are not tech savvy at all and they don't realize that if you try to draw a million locations on the map, it's going to be slow, no matter what you do. They try to request a report for multiple devices and the app hangs or becomes very unresponsive and they complain.

Maybe we can hide it behind some configuration parameter, so advanced users can have access to it.

**Comment by interduo** [comment]
Good point but there would be always some situations that You could generate big data requests. For this we could have setting called: maximum_selected_device (default: 10). And I think it would be suitable for tech and non-tech.

**Comment by interduo** [comment]
Is there any hope that this issue would be resolve?

**Comment by tananaev** [comment]
We have it enabled for some reports.

**Comment by interduo** [comment]
> We have it enabled for some reports.

But not in most important place - replay module :(

**Comment by tananaev** [comment]
Actually you're bringing up a good point. How would it even work if devices send positions at completely different times.

**Comment by interduo** [comment]
> Actually you're bringing up a good point. How would it even work if devices send positions at completely different times.

In most situations it doesnt matter if device location is +/-5 min.
We just want to know where they are in that time.
Well this is real time map substitution but for the selected time.

**Comment by tananaev** [comment]
What if it not 5 minutes, but 5 hours?

It was never available on any official version.

**Comment by interduo** [comment]
Situation 1: device was not available in that time (turned off LTE tracker)
show last device location (before that time on map),
mark it as red,
after mouse over present last location time,

Situation 2: the device was never available before (for newly added devices)
don't present on map

**Comment by interduo** [comment]
@tananaev did something move on this issue?

**Comment by interduo** [comment]
In 6.2 I do not have option REPLAY in Reports. Do I need to turn on it somewhere?

We can choose more than one device in reports - issue is partly resolved.

**Comment by interduo** [comment]
It is only implemented in legacy view.
Please implement that function in modern view.

**Comment by tananaev** [comment]
In recent versions most reports allow multi-device selection.
